optimum/onnxruntime/modeling_ort.py

```diff
@@ -428,6 +422,12 @@ def _from_pretrained(
         if model_save_dir is None:
             model_save_dir = new_model_save_dir
 
+        if provider == "TensorrtExecutionProvider" and use_io_binding:
+            logger.warning(
+                "There is no need to do IO binding for TensorrtExecutionProvider, `use_io_binding` is set to False."
```

Comment:
I would rather put a `logger.info` here, or better, show a log only if the user explicitly passed `use_io_binding=True, provider="TensorrtExecutionProvider"`.

Couldn't we make `use_io_binding = None` as default, and rather raise an error if it's set and the execution provider is not `CUDAExecutionProvider`?